### ISSUE TITLE
get RSSI

### ISSUE THREAD

**Issue by litong1314** [pauldemarco/flutter_blue#86]
is it possible to get RSSI of a CONNECTED device?

**Comment by GauravPatni** [comment]
Need help on same issue .... How to get RSSI value ?

**Comment by alextekartik** [comment]
Android (https://developer.android.com/reference/android/bluetooth/BluetoothGatt#readRemoteRssi()) seems to support it. and I guess iOS. I could help in the implementation (PR) if needed/wanted/accepted

**Comment by LiveRock** [comment]
Yes, I need to get RSSI for connected device.

**Comment by thomasgarrison** [comment]
Had anyone made progress on this? I can jump in and help if needed

**Comment by icaglar** [comment]
Yes I need help on same

**Comment by icaglar** [comment]
You can watch rssi values like below
FlutterBlue.instance.scanResults.listen((sr){
            sr.forEach((f){
               print("Scan Results:"+f.device.id.id.toString()+" - "+ f.rssi.toString());
            });
             
          });

**Comment by icaglar** [comment]
https://github.com/icaglar/flutter_blue
I created a pull request for this repository which provides getting rssi value of the connected device for ios
After clone this repository you have to recreate protos with the following command 
in protos folder
protoc --dart_out=../lib/gen/ ./flutterblue.proto 

After that you can access rssi value of the connected device  like this

 FlutterBlue.instance.connectedDevices.then((deviceList){
                  deviceList.forEach((d) async {
                    print("RSSI from device "+d.rssi.toString());
                  });
  });

**Comment by SurabhiS22** [comment]
> Had anyone made progress on this? I can jump in and help if needed

yes  pls , if possible can u pls help me in flutter bluetooth the RSSI part

**Comment by SurabhiS22** [comment]
> protoc --dart_out=../lib/gen/ ./flutterblue.proto

it doesn't worked , I did tried to recreate the protos but it says "Plugin failed with status code 127."

**Comment by kur0s4ki** [comment]
I have the same issue , I need to get the rssi value in Connected State. (On Android).
has anyone made any progression ?

**Comment by icaglar** [comment]
> > protoc --dart_out=../lib/gen/ ./flutterblue.proto
> 
> it doesn't worked , I did tried to recreate the protos but it says "Plugin failed with status code 127."

I think the issue related with your proto plugin. remove the plugin and install again

**Comment by jamal-Bahammou** [comment]
I have the same issue, I need to retrieve rssi from connected devices, need help ?

**Comment by Med0808** [comment]
same thing for me ; I need to access the rssi value on android (connected)

**Comment by icaglar** [comment]
I will share android side in a few days

**Comment by MouradElmehammedi** [comment]
i need to get the RSSI value on android any help?

**Comment by kur0s4ki** [comment]
> 
> 
> I will share android side in a few days

Thank you so much.

**Comment by icglr** [comment]
I have published android support .

**Comment by farhantandia** [comment]
where is the solution? I still couldn't find it :\

**Comment by chhamed** [comment]
is it possible to get RSSI without connecting to the device ?
